dapps/kiosk/src/Root.tsx

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Outlet } from 'react-router-dom';
+import { Toaster } from 'react-hot-toast';
+import { WalletKitProvider } from '@mysten/wallet-kit';
+import { Header } from './components/Base/Header';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { RpcClientContext } from './context/RpcClientContext';
+import { JsonRpcProvider, testnetConnection } from '@mysten/sui.js';
+
+export default function Root() {
+  const queryClient = new QueryClient();
+  const rpcProvider = new JsonRpcProvider(testnetConnection);
```

Comment:
Both this and the query client should be defined outside of the `Root` component.